### ISSUE TITLE
New devices are not being added to Home Assistant

### ISSUE THREAD

**Issue by rkkoszewski** [Vaskivskyi/ha-asusrouter#48]
New devices are not being added to Home Assistant and there seems to be no clear way of how to achieve this.

Based on the README I understand that new Devices are **not** created unless another integration creates it. Why is this? Is this a deliberate decision? Is there some manual workaround that could be used currently to make this work?

Also, if this is a deliberate decision, I'm aware that the device "spam" created in Home Assistant is annoying, but it is the official way that Home Assistant works. In the "System Options" of the integration you can switch off the "Enable newly added entities." flag which does the same as the current behavior. 

Any suggestions or ideas?

**Comment by Vaskivskyi** [comment]
Hey, @rkkoszewski,

The behaviour currently is the following. The integration creates `device_trackers` entities automatically, but does not create a HA device for each `device_tracker`. The entities are in the disabled state unless HA knows the device from another integration. If a HA device exists, `device_tracker` is added to it in the enabled state.

Do you suggest creating a HA device for each of the connected to the router device? In this case, there will be lots of devices with 1 entity only.

I was thinking about such an idea but in order to give users the possibility to have something like `internet` switches for each device to control their internet access from HA. Then devices would have 2 entities and it does already worth creating HA devices.

In any case, I am open to discussions and suggestions

**Comment by rkkoszewski** [comment]
Oww, I didn't notice that the entities are still added but without the device (Those entities are disabled by default, and can be enabled, which is the way to enable tracking for those devices). From what I was able to find out that is the "Official" way of how Home Assistant should handle these situations as of 2022 (Source: [Home Assistant: Improved handling of device tracker entities](https://www.home-assistant.io/blog/2022/02/02/release-20222/#improved-handling-of-device-tracker-entities)

Creating disabled "Devices" does not seem to be the official way to handle this situation. But for the "internet" switches idea, I'd suggest to create the "device + controls" when the tracking entity is enabled. 

Closing this issue, as the functionality I mentioned is working as intended. 
@Vaskivskyi  Thank you for clarifying all answers.

**Comment by Vaskivskyi** [comment]
Hey,

You have reopened this issue. I would need some more info to be able to help you. Did anything change with new devices adding after the last version release?

**Comment by rkkoszewski** [comment]
I'd like to reopen this issue to further discuss if there is no better approach to create "Devices" in Home Assistant. Currently the Home/NoHome tracking is added to an existing "Device" if it exists from another integration, but no new "Devices" are ever created by "ha-asusrouter". All other trackers can be found and enabled from the "Entities" section and be used there to track devices as "Entities".

The issue I came across is during the creation of automations. The handling of "Devices" in Home Assistant is pretty well fleshed out, while the usage of "Entities" seems inconsistent and error prone. You need to manually write the transition values by hand, and worst of all, it does not seem to be possible to use the "Entity" state in "If-Then" statements (Only the Entity attributes, but not the Entity state itself). This works perfectly fine with "Devices" where you can even select the state as dropdown. This seems to me like a bug in the Home Assistant Automation.

So now when you work on creating Automations in Home Assistant, we have "Devices" and other trackers that are "Entities" rather than "Devices" because no other integration has created those "Devices". That seems to be a somewhat inconsistent situation. 

I also re-read the official statement that I linked before ([Home Assistant: Improved handling of device tracker entities](https://www.home-assistant.io/blog/2022/02/02/release-20222/#improved-handling-of-device-tracker-entities)) and noticed that it only mentions that the devices that do not exist from other integrations, should appear as "Disabled" "Entities", but it does not mention what should happen when you "Enable" such an "Entity". So I think that part is free for interpretation, as I was not able to find further documentation on this.

**So TLDR:**

My suggestion is that if an "Entity" that is "Disabled" and has no "Device" because no other integration has created it, that if this "Entity" gets "Enabled" by the user, it should create a "Device" for that "Entity". That way the "Devices" would be consistent across all tracked devices.

What do you think @Vaskivskyi ? (Or anybody else has a better idea, or sources of how this should officially work?)

**Comment by Vaskivskyi** [comment]
I would agree with the TLDR. In general, I don't like entities which are not part of any device or service

Unfortunately, this is something, that should be implemented on the HA core side. But yes, I would prefer to get new devices, instead of just entities.

**Comment by rkkoszewski** [comment]
Oww, I thought that it's the integration that creates the devices. So it's not possible to create "Devices" from an integration? It is created by Home Assistant?

**Comment by Vaskivskyi** [comment]
Integration can easily create devices for every device tracked. I am not sure, how to handle the creation of the devices only when `device_tracker` entities are enabled.

But I'll look into the question. Maybe, it can be done

**Comment by Vaskivskyi** [comment]
It's probably a bit too late and I don't know whether you, @rkkoszewski, are still using AsusRouter, but with #762 users will now be able to automatically create devices when they enable device_tracker entities.

Will be released as soon as I finish 1 more feature for the next update

**Comment by Vaskivskyi** [comment]
Released in `0.28.0`
